src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class DefaultRazorIRLoweringPhase : RazorEnginePhaseBase, IRazorIRLoweringPhase
+    {
+        protected override void ExecuteCore(RazorCodeDocument codeDocument)
+        {
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            ThrowForMissingDependency<RazorSyntaxTree>(syntaxTree);
+
+            var visitor = new Visitor();
+
+            visitor.VisitBlock(syntaxTree.Root);
+
+            var irDocument = (DocumentIRNode)visitor.Builder.Build();
+            codeDocument.SetIRDocument(irDocument);
+        }
+
+        private class Visitor : ParserVisitor
+        {
+            public Visitor()
+            {
+                Builder = RazorIRBuilder.Document();
+
+                Namespace = new NamespaceDeclarationIRNode();
+                NamespaceBuilder = RazorIRBuilder.Create(Namespace);
+                Builder.Push(Namespace);
+
+                Class = new ClassDeclarationIRNode();
+                ClassBuilder = RazorIRBuilder.Create(Class);
+                Builder.Push(Class);
+
+                Method = new MethodDeclarationIRNode();
```

Comment:
Discussed this in person. I don't like execute because it's too much like MVC. I like better that we'd call it something neutral like `RazorMethodDeclarationIRNode`. That makes it clear that it's a "Razor" method without tying it to a framework